test/test_distributions.py

```diff
@@ -214,15 +228,16 @@ def _check_sampler_sampler(self, torch_dist, ref_dist, message, multivariate=Fal
         if isinstance(torch_samples, Variable):
             torch_samples = torch_samples.data
         torch_samples = torch_samples.cpu().numpy()
-        ref_samples = ref_dist.rvs(num_samples)
+        ref_samples = ref_dist.rvs(num_samples).astype(np.float64)
         if multivariate:
             # Project onto a random axis.
             axis = np.random.normal(size=torch_samples.shape[-1])
             axis /= np.linalg.norm(axis)
             torch_samples = np.dot(torch_samples, axis)
             ref_samples = np.dot(ref_samples, axis)
         samples = [(x, +1) for x in torch_samples] + [(x, -1) for x in ref_samples]
-        samples.sort()
+        shuffle(samples)  # necessary to prevent stable sort from making uneven bins for discrete
```

Comment:
Oh, this wasn't intended to work for discrete distributions. I see you're implementing a different test for those.